### PR TITLE
Refs [Task][30731] Create UI(Sign In Screen)

### PR DESCRIPTION
**Pull request by hieuna-441907**
## Related Tickets
- [#Task 30731: Create UI(Sign In Screen)](https://edu-redmine.sun-asterisk.vn/issues/30731)

## WHAT
- Create UI(Sign In Screen)
## HOW
- Create UI(Sign In Screen)
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (HuyTQ)
--- | --- | --- | --- | --- | ---
General |Code should be self-explanatory. | Get a feel of story reading, while going through the code. Use appropriate name for variables, functions and classes. If you are taking more time to understand the code, then either code needs refactoring or at least comments have to be written to make it clear. | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
General |Is this function or class too big? If yes, is the function or class having too many responsibilities? | | NO |<li>- [x] no</li>|<li>- [x] no</li>  
General | Does the code follow DRY?: Do not Repeat Yourself | Is there any redundant or duplicate code? DRY (Do not Repeat Yourself) principle: The same code should not be repeated more than twice. Consider reusable services, functions and components. Consider generic functions and classes. | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
General |Exception handling and cleanup (dispose) resources | | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
General | Does the code build and execute correctly? | | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
Conventions | Does the source code have end-line errors? | | NO |<li>- [x] no</li>|<li>- [x] no</li>  
Conventions | No hard coding, use constants/configuration values | | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [x] yes</li>|<li>- [x] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -4,4 +4,14 @@
...
+    <string name="title_login">Sign In to continue!</string>
+    <string name="title_welcome">Welcome,</string>
+    <string name="msg_hint_email">Email</string>
+    <string name="msg_hint_password">password</string>
+    <string name="title_login_button">Login</string>
+    <string name="msg_login_description">"I'm a new user, "</string>
+    <string name="msg_login_description_route">Sign up</string>
+    <string name="facebook_app_id">3372377532841706</string>
+    <string name="fb_login_protocol_scheme">fb3372377532841706</string>
+    <string name="label_sign_in_button">Sign In with Google</string>
```
Comment (NguyenTaTrung) [inline app/src/main/res/values/strings.xml]: Tách ra cho dễ nhìn ông

File: app/src/main/res/values/dimens.xml [hunk 2]
```diff
@@ -1,4 +1,8 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
     <dimen name="dp_4">4dp</dimen>
+    <dimen name="dp_8">8dp</dimen>
+    <dimen name="sp_10">10sp</dimen>
+    <dimen name="dp_16">16dp</dimen>
+    <dimen name="sp_16">16sp</dimen>
```
Comment (NguyenTaTrung) [inline app/src/main/res/values/dimens.xml]: sp và dp tách ra ông

File: app/src/main/res/layout/activity_sign_in.xml [hunk 3]
```diff
@@ -0,0 +1,142 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".ui.auth.SignInActivity">
```
Comment (NguyenTaTrung) [inline app/src/main/res/layout/activity_sign_in.xml]: remove thuộc tính tools:context

File: app/src/main/res/layout/activity_sign_in.xml [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        style="@style/Widget.Demo.TextInputLayout"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent">
+
+        <com.google.android.material.textfield.TextInputEditText
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
```
Comment (NguyenTaTrung) [inline app/src/main/res/layout/activity_sign_in.xml]: thêm id vào ông kiểu gì cũng dùng mà

File: app/src/main/res/layout/activity_sign_in.xml [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        app:errorEnabled="true"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent">
+
+        <com.google.android.material.textfield.TextInputEditText
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
```
Comment (NguyenTaTrung) [inline app/src/main/res/layout/activity_sign_in.xml]: thêm id

File: app/src/main/res/layout/activity_sign_in.xml (hunk 3 above)
Comment (thonc-2135) [reply app/src/main/res/layout/activity_sign_in.xml]: tại sao phải remove nhỉ

File: app/src/main/res/layout/activity_sign_in.xml (hunk 4 above)
Comment (thonc-2135) [reply app/src/main/res/layout/activity_sign_in.xml]: nếu dùng databinding thì đâu thiết cần id, tùy cái thôi

**Pull request merged**